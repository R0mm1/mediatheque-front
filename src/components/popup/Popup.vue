<template>
    <div class="med_popup">
        <div class="popup_header">
            <slot name="popup_header"></slot>
        </div>
        <div class="popup_body">
            <div class="popup_loader" :class="{no_loading: loaded}">
                <Loader></Loader>
            </div>
            <slot name="popup_body"></slot>
        </div>
        <div class="popup_footer">
            <slot name="popup_footer"></slot>
        </div>
    </div>
</template>

<script>
    import Loader from "../widgets/Loader";

    export default {
        name: "popup",
        components: {Loader},
        props: {
            loaded: {type: Boolean, default: false}
        }
    }
</script>

<style scoped lang="scss">
    .med_popup {
        display: flex;
        flex-direction: column;
        position: absolute;
        z-index: 3;
        width: calc(100% - 20px);
        height: calc(100% - 20px);
        margin: 10px;
        box-shadow: 1px 1px 5px black;
        background-color: #f8f5ef;
    }

    .popup_header {
        height: 4rem;
        display: flex;
    }

    .popup_body {
        position: relative;
        flex: 1;
        display: flex;
        overflow: hidden;
    }

    .popup_footer {
        height: 2.5rem;
        padding: 5px;
        display: flex;
        justify-content: flex-end;
    }

    .popup_loader{
        position: absolute;
        width: 100%;
        height: 100%;
        background-color: #f8f5ef;
        z-index: 1;
        text-align: center;
        transition: opacity .3s;

        &.no_loading{
            opacity: 0;
            visibility: hidden;
        }
    }
</style>
