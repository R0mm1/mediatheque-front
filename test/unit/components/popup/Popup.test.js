import {mount} from '@vue/test-utils'
import Popup from "@/components/popup/Popup.vue";

describe('Popup', () => {
    const wrapper = mount(Popup, {
        propsData: {
            loaded: false
        },
        slots: {
            popup_header: '<div>This is the header</div>',
            popup_body: '<div>This is the body</div>',
            popup_footer: '<div>This is the footer</div>'
        }
    });

    it('renders correctly when not loaded', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders correctly when loaded', () => {
        wrapper.setProps({
            loaded: true
        });
        expect(wrapper.element).toMatchSnapshot();
    })
});
