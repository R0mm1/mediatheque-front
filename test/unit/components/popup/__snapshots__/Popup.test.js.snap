// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup renders correctly when loaded 1`] = `
<div
  class="med_popup"
>
  <div
    class="popup_header"
  >
    <div>
      This is the header
    </div>
  </div>
   
  <div
    class="popup_body"
  >
    <div
      class="popup_loader"
    >
      <img
        src="/images/loaders/b.svg"
      />
    </div>
     
    <div>
      This is the body
    </div>
  </div>
   
  <div
    class="popup_footer"
  >
    <div>
      This is the footer
    </div>
  </div>
</div>
`;

exports[`Popup renders correctly when not loaded 1`] = `
<div
  class="med_popup"
>
  <div
    class="popup_header"
  >
    <div>
      This is the header
    </div>
  </div>
   
  <div
    class="popup_body"
  >
    <div
      class="popup_loader"
    >
      <img
        src="/images/loaders/b.svg"
      />
    </div>
     
    <div>
      This is the body
    </div>
  </div>
   
  <div
    class="popup_footer"
  >
    <div>
      This is the footer
    </div>
  </div>
</div>
`;
